### PR TITLE
[IMP] mail: allow dialog close by clicking on backdrop and with ESC

### PR DESCRIPTION
**Pull request by xdu-odoo**
https://www.odoo.com/web#id=2244203&action=3206&active_id=1519&model=project.task&view_type=form&cids=1&menu_id=4720
task-2244203

### CODE REVIEW

File: addons/mail/static/src/messaging/component/dialog/dialog.js
Comment (alexkuhn) [inline addons/mail/static/src/messaging/component/dialog/dialog.js]: why capture and not bubble?

File: addons/mail/static/src/messaging/component/dialog/dialog.js
Comment (seb-odoo) [inline addons/mail/static/src/messaging/component/dialog/dialog.js]: what is the default behavior of Esc that we want to prevent?

File: addons/mail/static/src/messaging/component/dialog/dialog.js
Comment (seb-odoo) [inline addons/mail/static/src/messaging/component/dialog/dialog.js]: Specify target `_onKeydownDocument`

**Comment by seb-odoo** [comment]
I tried on your runbot but I can't get a dialog to open for some reason?

**Comment by xdu-odoo** [comment]
> I tried on your runbot but I can't get a dialog to open for some reason?

You probably tried on community so it's normal, service managers are not mounted on community (it's a known bug : https://www.odoo.com/web#id=2244696&action=3206&active_id=1519&model=project.task&view_type=form&cids=1&menu_id=4720 )

**Review by seb-odoo — APPROVED** [review]
Ok, tested with debug assets, seems to work fine.

@alexkuhn final word on this one?

**Review by alexkuhn — APPROVED** [review]

**Comment by seb-odoo** [comment]
Oh we actually forgot to test the feature. But it's not a big deal, right now there are higher priorities than a test for this. This can be merged when green.

**Pull request merged**
